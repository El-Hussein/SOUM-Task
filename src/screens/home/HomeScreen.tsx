// In App.js in a new project

import {
  getHomeBrands,
  getHomeCategories,
  getHomeOffers,
  getHomeProducts,
} from '@actions';
import {
  selectAllOffers,
  selectBrands,
  selectCategories,
  selectProducts,
} from '@selectors';
import * as React from 'react';
import {View, Text} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';

const HomeScreen = () => {
  const dispatch = useDispatch();
  const homeCategories = useSelector(selectCategories);
  console.log('homeCategories', homeCategories);
  const homeBrands = useSelector(selectBrands);
  console.log('homeBrands', homeBrands);
  const homeProducts = useSelector(selectProducts);
  console.log('homeProducts', homeProducts);
  const homeOffers = useSelector(selectAllOffers);
  console.log('homeOffers', homeOffers);
  React.useEffect(() => {
    dispatch(getHomeCategories());
    dispatch(getHomeBrands(1));
    dispatch(getHomeProducts(1));
    dispatch(getHomeOffers(1));
  }, []);
  return (
    <View
      style={{
        flex: 1,
        backgroundColor: 'red',
        alignItems: 'center',
        justifyContent: 'center',
      }}>
      <Text>Home Screen</Text>
    </View>
  );
};

export default HomeScreen;
