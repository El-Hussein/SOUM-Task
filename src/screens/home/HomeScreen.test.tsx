import React from 'react';
import {render} from '@testing-library/react-native';
import HomeScreen from './HomeScreen';
// describe HomeScreen, it should render correctly and have a text of Home Screen

describe('HomeScreen', () => {
  it('should render correctly', () => {
    const tree = render(<HomeScreen />);
    expect(tree).toMatchSnapshot();
  });
  it('should have a text of Home Screen', () => {
    const {getByText} = render(<HomeScreen />);
    expect(getByText('Home Screen')).toBeTruthy();
  });
});
