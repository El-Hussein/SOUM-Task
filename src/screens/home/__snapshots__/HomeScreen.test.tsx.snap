// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "red",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Home Screen
  </Text>
</View>
`;
